modules/internal/ChapelDistribution.chpl

```diff
@@ -596,6 +596,18 @@ module ChapelDistribution {
               " (expected to be within ", parentDom, ")");
     }
 
+    proc canDoDirectAssignment(rhs: domain) {
+      if isRectangularDom(this.parentDom) &&
+         isRectangularDom(rhs.parentDom) {
```

Comment:
Are these tests for a rectangular parent domain out of conservatism, or required?  I.e., I'd expect that if the RHS was a subset of the LHS' parent domain, the direct assignment would be legal regardless of whether the parent domains were rectangular or not.

Reply:
Right.

The check for rhs is required simply because `isSubset` is not supported on sparse receivers. (I don't see any obvious reason that it isn't and I think it is just an unimplemented feature)

The check on `this` is not required, but I added it out of abundance of caution.

Are you OK with that, or do you want me to remove that check?

Reply:
I'm OK with that, just wanted to make sure I understood.  And if the parent domain isn't rectangular, it just fails this test and uses the fallback/slower assignment, is that right?  If so, I'm good with merging this, and thanks for getting it done.

Reply:
Yes, that's right. This function is called from within dsiAssignDomain, which has an `else` block that does the one-by-one logic.